Update to 1.21.11

BluSpring opened the pull request:
Closes #82.

This PR still uses Yarn mappings, and hasn't been migrated to Mojmap yet, I'll be leaving that up to Kira to do instead.
Fully implements Sodium's new config API and seems to function quite well from the limited testing I have done.
Also uses Sodium's official maven repository rather than pointing at Modrinth.
Haven't tested Neo + Sodium yet, but Neo alone works.

BluSpring committed:
Update to 1.21.11

Kira-NT commented:
Hey! Thanks **a lot** for the help! I took a brief look at all the Sodium changes and decided that nope, I don't have enough emotional bandwidth do deal with it right now :^

Kira-NT committed:
Leave the default filter mode alone

Creating this mixin was a mistake, and I'm not even sure why I went with it in the first place.

Fixes #68

Kira-NT committed:
Fix the "Fullscreen Resolution" option

Kira-NT committed:
`SodiumCompatEntrypoint` -> `SodiumCompat`

Kira-NT committed:
Fix video mode not being applied
